src/solvers/solver_hardness.cpp

```diff
@@ -91,10 +93,20 @@ void solver_hardnesst::register_clause(const bvt &bv)
 {
   current_hardness.clauses++;
   current_hardness.literals += bv.size();
+  std::vector<int> clause;
```

Comment:
`int`? or do we want a type with a defined length like `int_64`?

Reply:
❓ What are the memory and performance implications of storing a copy of the clauses?

Reply:
I haven't explored this yet. Although a much more efficient approach would be to store some form of index (specifically an index interval) to some existing clause storage, but I am not sure if this can be done.

Reply:
What are the metrics that you want to compute based on the clauses?

Reply:
To identify which program lines contribute to the SAT core. So given this infrastructure we can map clauses from the SAT core directly to LOC.

Reply:
What is a SAT core?

Reply:
Referring to unsatisfiable core here.
https://en.wikipedia.org/wiki/Unsatisfiable_core

Reply:
How are you going to compute the UNSAT core?

Reply:
There are tools which compute the UNSAT core given a CNF formula and DRAT proof of unsatisfiability (these proofs have become a standard in the SAT competitions). CBMC is used to get the dimacs CNF formula (--dimacs) and the clause to LOC mapping with --write-solver-stats-to option.

Reply:
If this can only be used wiith dimacs couldn't you just store the index (eg. line of the clause) in the dimacs output?

Reply:
How do I access the dimacs clause index here? Also is the dimacs output ordered?
I will need to map lines of program code to indices in the dimacs output.

Reply:
You can make `dimacs_cnft` (or probably `cnf_clause_listt`) implement `hardness_collectort`(in the same way other sat solvers implement that interface). It seems that the clauses are stored in order in `cnf_clause_listt`, so `dimacs_cnft` will print them in the same order as you get the `register_clause` calls. Of course, you'll have to test and confirm that.

Reply:
I now keep track of the clause counter in `satcheck_minisat2_baset<T>::lcnf` that calls `register_clause` and store the counter value rather than the clause itself in solver hardness. Have added optional DEBUG statements to verify if these match corresponding lines in the dimacs output.